plugin/wp-cli-login-server.php

```diff
@@ -155,7 +156,9 @@ private function validate(Magic $magic)
             throw new InvalidUser('No user found or no longer exists.');
         }
 
-        if (! $magic->private || ! wp_check_password($this->signature($user, $magic->redirect_url), $magic->private)) {
+        if (! $magic->private
+            || ! hash_equals(wp_hash($this->signature($user, $magic->redirect_url)), $magic->private)
```

Comment:
The arguments are in the wrong order. 

It's hash_equals(known_string, user_string)

https://www.php.net/manual/en/function.hash-equals.php

Reply:
You should delete a loaded magic link on hash mismatch

Reply:
Updated 👍